src/Symfony/Component/Serializer/Type/Type.php

```diff
@@ -0,0 +1,588 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Serializer\Type;
+
+use Symfony\Component\Serializer\Exception\InvalidArgumentException;
+use Symfony\Component\Serializer\Exception\LogicException;
+use Symfony\Component\Serializer\Exception\UnsupportedException;
+
+/**
+ * @author Mathias Arlaud <mathias.arlaud@gmail.com>
+ *
+ * @experimental in 7.0
+ */
+final class Type implements \Stringable
```

Comment:
This probably needs a description, indeed this class helps handling generics

Reply:
maybe add some syntaxic sugar?
```php
Type::createGeneric(string $mainType, string $subtractedType);
```

Reply:
Indeed, could be great @nikophil, we can even use variadics with something like:
```php
Type::generic(string $mainType, Type|string $subTypeOne, Type|string $subTypeTwo, ...)
```

Reply:
yes, I was actually thinking it could take a type in input